crates/agglayer-clock/src/block.rs

```diff
@@ -69,6 +72,12 @@ impl<P> BlockClock<P> {
         }
     }
 
+    /// Reinitialize the current Epoch number based on the current Block height.
+    fn reinitialize_epoch_number(&mut self, current_block: u64) {
+        let current_epoch = self.calculate_epoch_number(current_block);
+        self.current_epoch.store(current_epoch, Ordering::SeqCst);
```

Comment:
Seems error prone, and possibly racy, to keep both the current block and current epoch in sync. Could we just keep track of the current block and calculate the epoch from that on demand?

Reply:
This method is only used when starting the clock, I don't expect much risk on this. The `SeqCst` provides some guarantee too.

After that, only the `clock` that is spawn is updating the different counter (both `block` and `epoch`). The `ClockRef` which can be cloned and move from thread to thread has a ref to those counter but only read them to provide method such as `current_epoch` and `current_block_height`.

Those counter are not exposed outside the scope of the `clock` module, thus can't be read nor update by any other process. 

I agree that it could be better to have the epoch calculated on the flight, I'll create a ticket to address that.
One question that I have is, how could we properly deal with a change of epoch duration ?

Reply:
Here it is: #375